src/getShallowDiffs.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Pluck out the differences between previous/next props/state.
+ * Useful for debugging component update/renders.
+ *
+ * @param {Object} previous previousProps/previousState
+ * @param {Object} next nextProps/nextState
+ * @returns {Object} The differences
+ */
+function getShallowDiffs(previous: Object, next: Object): Object {
+  const diffs = Object.keys(previous).reduce((diffs, key) => {
+    if (previous[key] !== next[key]) {
+      return [...diffs, key]
+    }
+    return diffs
+  }, [])
```

Comment:
Couldn't we just use the simpler `.filter` method here instead? And shouldn't the return type be an array? Perhaps I'm missing something, but isn't this functionally equivalent to:

```
function getShallowDiffs(previous: Object, next: Object): Array {
  return Object.keys(previous).filter(key => key !== next[key])
}
```

Reply:
Oops! I merged it before I saw your question! sorry :D.

The output I have is this:

```
{
  diffs: ['propName'],
  previous: { propName: 'a' },
  next: { propName: 'b' },
}
```

From using it a bit.. I found that this structure was the most helpful. Hope that makes sense!

Reply:
Yeah, so the return type of `getShallowDiffs` should be an array. And I think the filter approach would be more concise.

Reply:
Ooops!! I misread. My bad 😅 

You are correct! I'll make the update